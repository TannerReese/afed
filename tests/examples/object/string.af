# Test string operations

"Hello" + "World" = ``,
"baz" + (foo + "bar") = ``,
3 * (foo + "baz") = ``,
"Hello, " * 4 = ``,
foo: "foo",

# String methods
"This is several chars".len = ``,
foo2: foo.lower + " vs. " + foo.upper = ``,
foo2 4 = ``,

# Polymorphic application
["Hello", [1, "a", true]].map \x: x.len = ``,

# Unimplemented operators
-"Hello" = ``,
"a" * "b" = ``,

