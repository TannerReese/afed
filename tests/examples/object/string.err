line 18, column 12 Eval Error: Unary operator - not implemented for type string
line 19, column 13 Eval Error: Binary operator * not implemented between types string and string
2 Eval Errors encountered
