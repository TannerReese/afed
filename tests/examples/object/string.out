# Test string operations

"Hello" + "World" = `"HelloWorld"`,
"baz" + (foo + "bar") = `"bazfoobar"`,
3 * (foo + "baz") = `"foobazfoobazfoobaz"`,
"Hello, " * 4 = `"Hello, Hello, Hello, Hello, "`,
foo: "foo",

# String methods
"This is several chars".len = `21`,
foo2: foo.lower + " vs. " + foo.upper = `"foo vs. FOO"`,
foo2 4 = `"v"`,

# Polymorphic application
["Hello", [1, "a", true]].map \x: x.len = `[5, 3]`,

# Unimplemented operators
-"Hello" = `Eval Error: Unary operator - not implemented for type string`,
"a" * "b" = `Eval Error: Binary operator * not implemented between types string and string`,

